crates/once-map/src/lib.rs

```diff
@@ -46,41 +47,61 @@ impl<K: Eq + Hash, V: Clone, H: BuildHasher + Clone> OnceMap<K, V, H> {
     ///
     /// Will hang if [`OnceMap::done`] isn't called for this key.
     pub async fn wait(&self, key: &K) -> Option<V> {
-        let entry = self.items.get(key)?;
+        let notify = {
+            let entry = self.items.get(key)?;
+            match entry.value() {
+                Value::Filled(value) => return Some(value.clone()),
+                Value::Waiting(notify) => notify.clone(),
+            }
+        };
+
+        // Prepare to wait.
+        let mut notification = pin!(notify.notified());
+        notification.as_mut().enable();
```

Comment:
Are we sure this is necessary? We are using [`Notify::notify_waiters`](https://docs.rs/tokio/latest/tokio/sync/struct.Notify.html#method.notify_waiters), and it seems like that doesn't benefit from `Notified::enable`, since `notify_waiters` doesn't "store" permits.

Reply:
That is, I'm wondering if its superfluous (not deleterious).

Reply:
It is necessary, calling `notify.notified()` does nothing but construct the future lazily. `enable` is what actually registers the waiter.

Reply:
Actually I might be wrong on that.

Reply:
> The Notified future is guaranteed to receive wakeups from notify_waiters() as soon as it has been created, even if it has not yet been polled.

Reply:
Ah whoops, my comments came in after your last two, but I hadn't seen those.

Yeah I'm just going on docs. Not on implementation.

I'd be inclined to leave out superfluous things because it can make things more confusing, but if we don't know it's superfluous we could leave it out and see whether the deadlocks re-appear. Or if you're feeling more conservative, add a comment explaining why it's there even though a read of the docs suggests it isn't needed.

Reply:
Ah okay, Tokio pulls a little trick here by tracking how many times `notify_waiters` has been called. I missed that, you're correct in that we don't need the `enable`.

Reply:
I updated the comments to make it clear this only works with `notify_waiters`.